asmcomp/cmmgen.ml

```diff
@@ -2434,6 +2434,55 @@ and transl_letrec env bindings cont =
         fill_blocks rem
   in init_blocks bsz
 
+(* Insert instrumentation for afl-fuzz *)
+
+let afl_area_ptr = Cconst_symbol "caml_afl_area_ptr"
+let afl_prev_loc = Cconst_symbol "caml_afl_prev_loc"
+let afl_map_size = 1 lsl 16
+
+let rec with_afl_logging b =
+  if !Clflags.afl_inst_ratio < 100 && Random.int 100 >= !Clflags.afl_inst_ratio then instrument b else
+  let instrumentation =
+    let cur_location = Random.int afl_map_size in
+    let cur_pos = Ident.create "pos" in
+    let afl_area = Ident.create "shared_mem" in
+    Clet(afl_area, Cop(Cload Word_int, [afl_area_ptr]),
+    Clet(cur_pos,  Cop(Cxor, [Cop(Cload Word_int, [afl_prev_loc]); Cconst_int cur_location]),
+    Csequence(
+      Cop(Cstore(Byte_unsigned, Assignment),
+          [Cop(Cadda, [Cvar afl_area; Cvar cur_pos]);
+           Cop(Cadda, [Cop (Cload Byte_unsigned,
+                            [Cop(Cadda, [Cvar afl_area; Cvar cur_pos])]);
+                       Cconst_int 1])]),
+      Cop(Cstore(Word_int, Assignment),
+          [afl_prev_loc; Cconst_int (cur_location lsr 1)])))) in
+  Csequence(instrumentation, instrument b)
```

Comment:
Is there some document somewhere explaining the communication between the process and the tool ?
If it does, could you add a link in a comment (or if it is small enough directly put it here) ?